cm2350/e200z7.py

```diff
@@ -580,6 +583,38 @@ def getByteDef(self, va):
     def isValidPointer(self, va):
         return self.mmu.getDataEntry(va)[2] is not None
 
+    def readRegValue(self, reg):
+        """
+        A method to read a register value like getRegister() that automatically
+        determines if a register is a BitFieldSPR and therefore reads the
+        proper object instead of from the register file.
+
+        Because this is less efficient than getRegister() it should not be used
+        unless necessary.
+        """
+        # TODO: figure out how to integrate BitFieldSPR objects more seemlessly?
+        sprobj = self.sprs.get(reg)
+        if sprobj is None:
+            return self.getRegister(reg)
+        else:
+            return sprobj.read(self)
```

Comment:
nm.  Now that I'm looking at it, these are the vstruct-based storage mechanisms which are currently how the peripherals access the data.  In the future, this might be better done differently, but for now it works well.